src/lib/expression/evaluation/expression_evaluator.cpp

```diff
@@ -253,27 +254,97 @@ ExpressionEvaluator::_evaluate_in_expression<ExpressionEvaluator::Bool>(const In
   if (right_expression.type == ExpressionType::List) {
     const auto& array_expression = static_cast<const ListExpression&>(right_expression);
 
+    DebugAssert(!array_expression.elements().empty(), "IN clauses with an empty list are invalid");
```

Comment:
It's illegal and MySQL and PostgreSQL don't support it:
```
         <in predicate value> ::=
                <table subquery>
              | <left paren> <in value list> <right paren>

         <in value list> ::=
              <value expression> { <comma> <value expression> }...
```

I wouldn't mind, but handling that edge case caused issues further down the line.

Reply:
I dislike crippling the ExpressionEvaluator because the SQL standard (which the Expression system doesn't care about) doesn't allow a very specific path. Once we're past SQL-translation, every LQP/Expression that makes logical sense should be valid, without paying attention to what SQL thinks is valid or not.

Reply:
It's similar with our query plans: Do we care whether they correspond to valid SQL? No, and it would be confusing if we did.

Reply:
okay